### PR TITLE
Allow Python 3 To Be Built On Non-ARM Architectures

### PR DESCRIPTION
**Pull request by TheBrokenRail**
This makes it no longer hardcoded to build on ARM.

### CODE REVIEW

**Comment by inclement** [comment]
Thanks, would you mind fixing the minor style errors Travis has flagged up via flake8?

File: pythonforandroid/recipes/python3/__init__.py [hunk 1]
```diff
@@ -67,14 +67,20 @@ def build_arch(self, arch):
...
             # TODO: Get this information from p4a's arch system
-            android_host = 'arm-linux-androideabi'
+            android_host = arch.command_prefix
             android_build = sh.Command(join(recipe_build_dir, 'config.guess'))().stdout.strip().decode('utf-8')
-            platform_dir = join(self.ctx.ndk_dir, 'platforms', platform_name, 'arch-arm')
-            toolchain = '{android_host}-4.9'.format(android_host=android_host)
+            platform_dir = join(self.ctx.ndk_dir, 'platforms', platform_name, arch.platform_dir)
+            toolchain = '{android_host}-4.9'.format(android_host=arch.toolchain_prefix)
             toolchain = join(self.ctx.ndk_dir, 'toolchains', toolchain, 'prebuilt', 'linux-x86_64')
+
```
Comment (AndreMiras) [inline pythonforandroid/recipes/python3/__init__.py]: `W293 blank line contains whitespace`

File: pythonforandroid/recipes/python3/__init__.py [hunk 2]
```diff
@@ -67,14 +67,20 @@ def build_arch(self, arch):
...
-            toolchain = '{android_host}-4.9'.format(android_host=android_host)
+            platform_dir = join(self.ctx.ndk_dir, 'platforms', platform_name, arch.platform_dir)
+            toolchain = '{android_host}-4.9'.format(android_host=arch.toolchain_prefix)
             toolchain = join(self.ctx.ndk_dir, 'toolchains', toolchain, 'prebuilt', 'linux-x86_64')
+            
+            target_data = arch.command_prefix.split('-')
+            if target_data[0] == 'arm':
+                target_data[0] = 'armv7a'
+            target = '-'.join([target_data[0], 'none', target_data[1], target_data[2]])
+
```
Comment (AndreMiras) [inline pythonforandroid/recipes/python3/__init__.py]: `W293 blank line contains whitespace`

**Comment by AndreMiras** [comment]
Thanks for the PR, here is the linting error:
```
pythonforandroid/recipes/python3/__init__.py:75:1: W293 blank line contains whitespace
pythonforandroid/recipes/python3/__init__.py:80:1: W293 blank line contains whitespace
```
You can always run `tox` locally before PR to check that.

**Comment by TheBrokenRail** [comment]
Fixed Lint!

**Comment by inclement** [comment]
Sorry for the delay, I thought this had been merged already. Thanks again!

**Pull request merged**